### ISSUE TITLE
AM/PM requis pour l'heure sur Firefox Mac

### ISSUE THREAD

**Issue by SamBThor** [LAB-MI/attestation-deplacement-derogatoire-q4-2020#22]
Ce n'est pas une bonne idée pour une application française.

**Comment by viincenb** [comment]
C'est vrai que je trouve ça un peu bizarre à l'utilisation, je verrais bien un double select, le premier pour l'heure (format 24h), et le second pour les minutes (toutes les 5min).

**Comment by julienw** [comment]
Par hasard, es-tu sur un firefox en anglais, ou ton système est-il configuré en anglais ?

**Comment by bbenoit** [comment]
Note : bien que peu souvent utilisé en français, les termes _AM_ et _PM_ sont issus du latin pour _Ante Meridiem_ et _Post Meridiem_…

Voir https://fr.wikipedia.org/wiki/Syst%C3%A8me_horaire_sur_12_heures

**Comment by lautome** [comment]
> Ce n'est pas une bonne idée pour une application française.

C'est lié à ton système.

**Comment by a2br** [comment]
Lisez les autres issues sur les problèmes de dates. Je le répète : les sélecteurs de date / heure (sauf pour la date de naissance) sont natifs.

**Comment by viincenb** [comment]
Ah oui, my bad, je ne pensais pas que mon chrome ait un timepicker natif si mauvais...

**Comment by a2br** [comment]
Au temps pour moi, il y a peut être un problème dans ce cas. Pas sur la dernière version de Chrome / Windows en tt cas.

**Comment by FabienTregan** [comment]
> Lisez les autres issues sur les problèmes de dates. Je le répète : les sélecteurs de date / heure (sauf pour la date de naissance) sont natifs.

Ce n'est pas un argument. Un test rapide sur les 4 personnes de mon foyer et 3 personnes interrogées par telephone montre qu'une seule personne a réussi a générer l'attestation, et seulement apres avoir trouvé une version forkée sur laquelle le chant etait prerempli automatiquement.

Ceci doit etrecorrigé.

**Comment by zinoviosDev** [comment]
Aucun soucis avec Chrome on Windows configuré en français (et probablement avec Firefox non plus).
-> Configurez

> > Lisez les autres issues sur les problèmes de dates. Je le répète : les sélecteurs de date / heure (sauf pour la date de naissance) sont natifs.
> 
> Ce n'est pas un argument. Un test rapide sur les 4 personnes de mon foyer et 3 personnes interrogées par telephone montre qu'une seule personne a réussi a générer l'attestation, et seulement apres avoir trouvé une version forkée sur laquelle le chant etait prerempli automatiquement.
> 
> Ceci doit etrecorrigé.

Aucun soucis avec Chrome on Windows configuré en français (et probablement avec Firefox non plus).
-> Faites le test avec votre navigateur en français.

**Comment by a2br** [comment]
> > Lisez les autres issues sur les problèmes de dates. Je le répète : les sélecteurs de date / heure (sauf pour la date de naissance) sont natifs.
> 
> Ce n'est pas un argument. Un test rapide sur les 4 personnes de mon foyer et 3 personnes interrogées par telephone montre qu'une seule personne a réussi a générer l'attestation, et seulement apres avoir trouvé une version forkée sur laquelle le chant etait prerempli automatiquement.
> 
> Ceci doit etrecorrigé.

Oui, sans aucun doute, cela doit être corrigé si problème il y a, mais quoi qu'il en soit nous n'avons aucune info sur les configs concernées et ne pouvons donc pas le régler. Une recherche sur Google montre que ça dépend bel et bien de la configuratoin du navigateur / OS.

**Comment by a2br** [comment]
@FabienTregan C'est tout à fait un argument, la question est s'il est juste au faux.
Marche sur Chromium, Windows.

**Comment by Mathisca** [comment]
Il faudrait peut être ne pas utiliser le format natif du système alors...
C'est la première fois que je vois ce type de format pour un champ date et c'est très peu ergonomique.

**Comment by FabienTregan** [comment]
Quelques éléments supplémentaires :
-l'une des machien a un linux mint volontairement en anglais avec le firefox par defaut à jour
-L'autre est sous un windows 10 home en francais, firefox est installé en anglais (la personne qui l'a installé comme elle a pu ne sait pas pourquoi).

Dans les deux cas le champs date ressemble à ceci :
![image](https://user-images.githubusercontent.com/2663407/97960476-dc74ef80-1db1-11eb-8390-0ae2533311f1.png)

Une fois qu'on a compris qu'on pouvait modifier les champs avec les flèches haut / bas c'esst vraiment simple, mais par deffaut tous les utilisateurs avec qui j'ai testé on cu devoir entrer les secondes,on tapé des chiffres qui ont été ignorés

L'attestation produite utilise bien le format 24h dansles deux as (`9:00 PM` est imprimé et encodé dans le QR code comme `21:00`).

Il s'agit a prior bien d'un probleme de navigateur installé en anglais, sans que celà en fasse un point moins bloquant pour les utilisateurs concernés.
